lightning/src/ln/peer_handler.rs

```diff
@@ -242,6 +353,40 @@ macro_rules! encode_msg {
 	}}
 }
 
+impl<Descriptor: SocketDescriptor, CM: Deref, L: Deref> PeerManager<Descriptor, CM, DummyRouteHandler, L> where
+		CM::Target: ChannelMessageHandler,
+		L::Target: Logger {
+	/// Constructs a new PeerManager with the given channel message handler. No routing message
+	/// handler is used and network graph messages are ignored.
+	///
+	/// ephemeral_random_data is used to derive per-connection ephemeral keys and must be
+	/// cryptographically secure random bytes.
+	pub fn new_without_routing(channel_message_handler: CM, our_node_secret: SecretKey, ephemeral_random_data: &[u8; 32], logger: L) -> Self {
+		Self::new(MessageHandler {
+			chan_handler: channel_message_handler,
+			route_handler: DummyRouteHandler{},
+		}, our_node_secret, ephemeral_random_data, logger)
+	}
+}
+
+impl<Descriptor: SocketDescriptor, RM: Deref, L: Deref> PeerManager<Descriptor, DummyChannelHandler, RM, L> where
+		RM::Target: RoutingMessageHandler,
+		L::Target: Logger {
+	/// Constructs a new PeerManager with the given routing message handler. No channel message
```

Comment:
nit: I prefer for greppability to have the official struct name: i.e. s/channel message handler/ChannelMessageHandler, here and in `new_without_routing`

Reply:
Throwing out an alternative: `new_routing_only` and `new_channel_only`

Reply:
Ah I don't think the original comment ^ was resolved :)

Reply:
Oh lol, oops.